### PR TITLE
Fix schema initialization

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

There are a couple of race condition  when multiple producers consumer are trying to use a new topic with schema defined.

If the scheme is not stored yet, we would get ZK.NodeExists exception when multiple attempts to create the schema happen.

If the schema is already there, we can have errors with multiple open ledger operations on the same ledger.

An easy way to trigger this behavior is to create partitioned topics with schema definition.

### Modifications

Use map of futures to keep track of in progress operations and have all the request to piggyback on the same future.

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.Schema;
+import org.testng.annotations.Test;
+
+public class PartitionedTopicsSchema extends BrokerBkEnsemblesTests {
+
+    /**
+     * Test that sequence id from a producer is correct when there are send errors
+     */
+    @Test(invocationCount = 100, skipFailedInvocations = true)
```
Comment (rdhabalia) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java]: is it necessary to keep `invocationCount = 100` for real test?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+import org.apache.pulsar.broker.service.BrokerBkEnsemblesTests;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.Schema;
+import org.testng.annotations.Test;
+
+public class PartitionedTopicsSchema extends BrokerBkEnsemblesTests {
```
Comment (rdhabalia) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java]: should we rename it with post-fix `Test`

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java (hunk 1 above)
Comment (merlimat) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java]: ups, forgot about that. removing

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java (hunk 2 above)
Comment (merlimat) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/schema/PartitionedTopicsSchema.java]: yes

**Pull request merged**